inst/tests/testthat/test-helloworld.R

```diff
@@ -0,0 +1,34 @@
+library(testthat)
+library(methods)
+library(grpc)
+library(processx)
+
+context('helloworld example')
+
+## start a server in the background
+p <- process$new(commandline = "R -e \"demo('helloserver', 'grpc', ask = FALSE)\"", stdout = "|", stderr = "|")
+
+test_that('server started', {
+    expect_true(p$is_alive())
+})
+
+## wait for the server to start
+while (TRUE) {
+    Sys.sleep(1)
+    if (any(p$read_output_lines() == "Entering Event Loop")) {
+        break()
+    }
+}
+
+test_that('server running', {
+    expect_true(p$is_alive())
+})
+
+test_that('using client', {
+    expect_true(any(
+        grepl('Hello, Neal',
+              capture.output(demo('helloclient', 'grpc', ask = FALSE)))))
```

Comment:
`demo` is a wrapper around `source`, you can `source('helloclient.R', local=TRUE)` and actually look at / test the protobuf responses. 

If the goal is to test that the demos work end-to-end, let's use `expect_output_file`   instead of regexes. When I get the debugging macros put into server.cpp and client.cpp for #5, we should have  much more clean output to test against.

Reply:
Yeah, maybe setting up the connection and doing `grpc_client` would be cleaner :+1: 
I just wanted to get something done real quickly, and as that's being done in the demo, it made sense to run that :) `source` would be longer as I guess there's a need for `system.file(package = ...)`, so I'm not really into that :) But I leave it up to you, good points.